src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -662,11 +662,22 @@ class PersonAgent(
             netTripCosts // Again, includes tolls but "net" here means actual money paid by the person
           )
         )
+      var dataForNextLegOrActivity = data.copy(
```

Comment:
Rather than a `var`, how about a `val` that better illustrates this anyway:

```
val dataForNextLegOrActivity = if(...)
    //stuff
    data.copy(...)
else data.copy(...default...)
```